lib/internal/loader/ModuleRequest.js

```diff
@@ -36,21 +38,31 @@ loaders.set('esm', async (url) => {
 
 // Strategy for loading a node-style CommonJS module
 loaders.set('cjs', async (url) => {
-  return createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading CJSModule ${url}`);
-    const CJSModule = require('module');
-    const pathname = internalURLModule.getPathFromURL(new URL(url));
-    CJSModule._load(pathname);
+  debug(`Loading CJSModule ${url}`);
+  const CJSModule = require('module');
+  const pathname = internalURLModule.getPathFromURL(new URL(url));
+  const exports = CJSModule._load(pathname);
+  const es = Boolean(exports[esModuleSymbol] || exports.__esModule);
+  const keys = es ? Object.keys(exports) : ['default'];
+  return createDynamicModule(keys, url, (reflect) => {
+    if (es) {
+      for (const key of keys)
```

Comment:
There might be some edge cases where `module.exports` is a proxy, can you please check if that would cause problems?

Reply:
What's the performance of for/of loop compared to alternatives (see http://benediktmeurer.de/2017/09/07/restoring-for-in-peak-performance/)? cc @bmeurer .
I would go for an old-school for..in loop as performance here might be an issue.

Reply:
I think we should use a `for(;;)` loop here. They are faster.

Reply:
Object.keys with for loop is the fastest after 6.1, which is where this code will be running anyway